src/SimpleSoft.Database.EFCore/EFCoreUnitOfWork.cs

```diff
@@ -0,0 +1,239 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace SimpleSoft.Database
+{
+    /// <summary>
+    /// An implementation compatible with Entity Framework Core for the Unit of Work pattern.
+    /// </summary>
+    public class EFCoreUnitOfWork : IUnitOfWork
+    {
+        private readonly IServiceProvider _provider;
+
+        /// <summary>
+        /// Creates a new instance
+        /// </summary>
+        /// <param name="provider"></param>
+        public EFCoreUnitOfWork(IServiceProvider provider)
+        {
+            _provider = provider;
+        }
+
+        /// <inheritdoc />
+        public async Task<ITransaction> BeginTransactionAsync(CancellationToken ct)
+        {
+            var transaction = _provider.GetRequiredService<EFCoreTransaction>();
+            await transaction.BeginAsync(ct);
```

Comment:
If the method `BeginAsync` fails, the transaction should be disposed before throwing the exception of a memory leak may happen since the instance was created, but it won't be returned.

```csharp
var transaction = _provider.GetRequiredService<EFCoreTransaction>();
try {
  await transaction.BeginAsync(ct);
} catch {
  await transaction.DisposeAsync();
  throw;
}
return transaction;
```